c++/src/kj/common.h

```diff
@@ -1753,13 +1753,14 @@ class ArrayPtr: public DisallowConstCopyIfNotConst<T> {
     return other.size_ <= size_ && slice(0, other.size_) == other;
   }
 
-  inline Maybe<size_t> findFirst(char c) const {
-    const char* pos = reinterpret_cast<const char*>(memchr(ptr, c, size_));
-    if (pos == nullptr) {
-      return nullptr;
-    } else {
-      return pos - ptr;
+  inline Maybe<size_t> findFirst(const T& match) const {
+    for (int i = 0; i < size_; i++) {
+      // TODO(now): Do we need to consider if `ptr + i` == nullptr?
+      if (*(ptr + i) == match) {
```

Comment:
`*(ptr + i)` can also be written `ptr[i]` -- they mean the same thing but the latter is shorter.

Reply:
In fact, fun fact: `*(i + ptr)` is obviously the same as `*(ptr + i)`. And so `i[ptr]` is the same as `ptr[i]`.

Reply:
That feels so wrong but... also makes sense?

```
~ cat wat.c
#include <stdio.h>

int main() {
	int arr[] = {1, 2};
	int *p = arr;
	printf("%d\n", p[0]);
	printf("%d\n", 0[p]);
	
	if (p[1] == 1[p] && &p[1] == &1[p]) {
		printf("wtf\n");
	} else {
		printf("gcc disagrees\n");
	}
	return 0;
}
~ ./wat
1
1
wtf
~
```
:upside_down_face: